minio/api.py

```diff
@@ -1389,7 +1389,8 @@ def presigned_url(self, method,
                       object_name,
                       expires=timedelta(days=7),
                       response_headers=None,
-                      request_date=None):
+                      request_date=None,
+                      endpoint_url=None):
```

Comment:
Is it assumed the same credentials work for this endpoint? It just makes confusion for users. Can't you fix the issue at gateway side of minio server?

Reply:
Agreed @balamurugana these are ad-hoc use cases we shouldn't entertain. This library is simple as it is and serves it purpose. Generate presigned URL for the right URL with the right constructor is the way to go.

We should avoid cockpit-style options